api/v1beta1/network_disruption.go

```diff
@@ -98,6 +98,8 @@ type NetworkDisruptionHostSpec struct {
 type NetworkDisruptionServiceSpec struct {
 	Name      string `json:"name"`
 	Namespace string `json:"namespace"`
+	// +optional
+	Ports []int `json:"ports,omitempty"`
```

Comment:
Can't really add kubebuilder validation in there. what do you think? Because we check at the validation that the port exists in the service I personally think it's fine, but I could be wrong.

Reply:
If we add protocol as well let's do this then, thanks!

Reply:
So if I take your example with `protocol`:
```yaml
services:
  - name: demo # service name
    namespace: chaos-demo # service namespace
    port: 8080 # optional. affected port. No port means all ports are affected
    protocol: udp
  - name: demo # service name
    namespace: chaos-demo # service namespace
    port: 8081 # optional. affected port. No port means all ports are affected
```

Instead of this:
```yaml
services:
  - name: demo # service name
    namespace: chaos-demo # service namespace
    ports:
       - port: 8080 # optional. affected port. No port means all ports are affected
         protocol: udp
       - port: 8081 # optional. affected port. No port means all ports are affected
```

You could also create a dedicated type being `ports: [1,2,3,4]` that would define the protocol to TCP as a default value 😄 

Both declaration seems fine to me, I may prefer the one closer to Kubernetes service definition (ease copy paste)

Reply:
I prefer to do this as well, i'll settle on your suggestion. Thanks Matthieu!